tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java

```diff
@@ -202,7 +202,7 @@ private static void extractResources() {
       LOG.info("Unpacking src directory..");
       Utils.unzipArchive(Constants.TONY_SRC_ZIP_NAME, "./");
     }
-    if (new File(Constants.PYTHON_VENV_ZIP).exists()) {
+    if (new File(Constants.PYTHON_VENV_ZIP).exists() && new File(Constants.PYTHON_VENV_ZIP).isFile()) {
```

Comment:
Can we factor `extractResources()` out into a utility class and have TaskExecutor and ApplicationMaster call it rather than duplicating code?

Reply:
Create a local variable to store `new File(Constants.PYTHON_VENV_ZIP)` rather than duplicating it.

Reply:
makes sense, updated